### PR TITLE
[TRAFODION-2044] Support SALT clause on CREATE TABLE LIKE

### PR DESCRIPTION
**Pull request by DaveBirdsall**
Formerly, several clauses (examples: SALT and STORE BY) were allowed on CREATE TABLE LIKE, but were ignored. So, for example, one could say, "CREATE TABLE t2 LIKE t1 SALT USING 6 PARTITIONS", but t2 would end up getting the same salting as t1 instead.

With this change, the parser has been changed so that, except for SALT, specifying these clauses will now result in a syntax error.

For the SALT clause, it is now fully supported with CREATE TABLE LIKE. So, in our example above, t2 would get 6 partitions, and would be salted on the primary key.

A new regression test, seabase/TEST040, has been added to cover this support. The manuals have also been updated, and a few unrelated corrections in CREATE TABLE have been made also.

This pull request also updates the expected results for regression test compGeneral/TEST015, which has been failing for a little while. This change is unrelated to the rest of the changes.

### CODE REVIEW

File: core/sql/parser/sqlparser.y [hunk 1]
```diff
@@ -24909,7 +24946,6 @@ create_table_as_token: TOK_AS
 
 /* type pElemDDL */
 table_definition_body : table_element_list
-                      | like_definition
                       | external_table_definition
```
Comment (robertamarton) [inline core/sql/parser/sqlparser.y]: It looks like the new like option is no longer part of a table_definition_body but as a separate high-level production.  Would this break some operations?  For example, I see on line 24594 that a "like" option is syntactically correct with <special_table_name> option.  Would this still be allowed with your change?

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 2]
```diff
@@ -2235,13 +2235,19 @@ session.
...
-    { table-spec | like-spec }
+    table-spec
     [SALT USING num PARTITIONS [ON (column[, column]...)]]
     [STORE BY {PRIMARY KEY | (key-column-list)}]
-    [HBASEOPTIONS (hbase-options-list)]
+    [HBASE_OPTIONS (hbase-options-list)]
     [LOAD IF EXISTS | NO LOAD]
     [AS select-query]
 
+CREATE [VOLATILE] TABLE IF NOT EXISTS table
```
Comment (robertamarton) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Not your issue but happened to notice it.  Isn't the IF NOT EXISTS clause optional?

File: core/sql/parser/sqlparser.y (hunk 1 above)
Comment (DaveBirdsall) [inline core/sql/parser/sqlparser.y]: This is intentional. The problem is that including like_definition in table_definition_body allows things like STORE BY and several other clauses to follow the LIKE option. But the LIKE code just ignores them all. That is, the original implementation was incomplete. I wanted to make the unsupported syntax invalid.

Comment (DaveBirdsall) [inline core/sql/parser/sqlparser.y]: Incidentally, I did a full run of all regression tests having CREATE TABLE LIKE over the weekend to make sure I didn't miss anything. They all passed.

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 2 above)
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Good catch! Will fix, either in a rework or a later check-in.

File: core/sql/regress/seabase/TEST040 [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+-- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+-- KIND, either express or implied.  See the License for the
+-- specific language governing permissions and limitations
+-- under the License.
+--
+-- @@@ END COPYRIGHT @@@
+
+set schema $$TEST_SCHEMA$$;
+log LOG040 clear;
+
```
Comment (robertamarton) [inline core/sql/regress/seabase/TEST040]: May want to add a "cleanup" first before starting the test.  This way, if this test failed previously, you are starting with a clean slate.

Comment (robertamarton) [inline core/sql/regress/seabase/TEST040]: As an alternative, you could just create a new schema and drop it with the cascade option.

Comment (DaveBirdsall) [inline core/sql/regress/seabase/TEST040]: Thanks; will fix. I copied this from another test.

**Comment by robertamarton** [comment]
Looks good

**Comment by DaveBirdsall** [comment]
Thanks for the careful review. Will submit a rework addressing the items you mentioned.

File: core/sql/parser/sqlparser.y (hunk 1 above)
Comment (robertamarton) [inline core/sql/parser/sqlparser.y]: This may be correct.  Not sure if the like options should be allowed for other create table productions.

**Comment by robertamarton** [comment]
Thanks Dave, looks good.

**Pull request merged**